### ISSUE TITLE
error handling w/ duplicate torrents

### ISSUE THREAD

**Issue by devTristan** [feross/webtorrent#348]
```javascript
var WebTorrent = require('webtorrent-hybrid');

var client = new WebTorrent();

client.seed('./www', torrent => {
  console.log('seed worked');
  try {
    client.add(torrent.infoHash, () => console.log('add worked'))
  } catch (err) {
    console.error('got error', err)
  }
})
```

Expected output:
```
$ node test.js 
seed worked
add worked
```

Or, less preferably:
```
$ node test.js 
seed worked
got error [Error: There is already a swarm with info hash 55f65714bb41dcdf19fd3ccceb89fdfded970de2 listening on port 48839]
```

Actual output:
```
$ node test.js 
seed worked
events.js:85
      throw er; // Unhandled 'error' event
            ^
Error: There is already a swarm with info hash ... listening on port 48839
    at /.../node_modules/webtorrent-hybrid/node_modules/webtorrent/node_modules/bittorrent-swarm/lib/tcp-pool.js:101:22
    at process._tickCallback (node.js:355:11)
```

Might it be worth adding an `err` argument to the callbacks, or just returning an existing torrent when it's already been added/seeded?

**Comment by feross** [comment]
You don't need to manually add the torrent to the client when you're seeding - it's handled for you. `client.seed` actually calls `client.add` internally :-)

Can't add an `err` argument to the callback because it's actually just a shorthand for listening to the 'torrent' event. Events don't have `err` arguments.

I like the idea of returning the existing torrent instead of failing with an error. Let's do that.

**Comment by devTristan** [comment]
> I like the idea of returning the existing torrent instead of failing with an error. Let's do that.

I was hoping you'd say that :)

I'm looking through the code. The error is thrown [here](https://github.com/feross/bittorrent-swarm/blob/master/lib/tcp-pool.js#L102), in bittorrent-swarm. I believe it can be caught and handled [here](https://github.com/feross/webtorrent/blob/master/index.js#L170).

I could intercept this error and emit the `torrent` event again, but that may cause unexpected behavior for anybody expecting that event to be emitted once per torrent, and that would sort of leak behavior out in a weird way which I don't like.

Instead, I think I'll iterate through `self.torrents`, running `clientOnTorrent` on each one, assuming `ontorrent` is has been defined. Since `clientOnTorrent` only runs `ontorrent` when the infoHash matches, I think that should do it... maybe...

**Comment by devTristan** [comment]
PR here https://github.com/feross/webtorrent/pull/351

**Comment by feross** [comment]
Released as 0.49.0.
